lib/parse.js

```diff
@@ -12,20 +15,33 @@ exports = module.exports = function(content, options) {
       root = exports.evaluate('<root></root>', options)[0];
 
   root.type = 'root';
+  root.parent = null;
 
   // Update the dom using the root
   exports.update(dom, root);
 
   return root;
 };
 
+function parseWithParse5 (content) {
+  var parseAsDocument = /^(\s|<!--.*?-->)*?<(!doctype|html|head|body)(.*?)>/i.test(content),
```

Comment:
If we'll always use fragment parsing parser will always omit `<html>`, `<body>` and `<head>` tags and doctypes, so it will basically will return empty AST for full documents. On the other hand, enabling full parsing will always add these tags. Even in 1.0 you will still need this hack.